### PR TITLE
Fix query counts

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/FTL/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 

## 10

---

Further fixes to ensure the independent quantities `dns_queries_today`, `dns_queries_all_types` and the newly introduced `dns_queries_all_replies` are indeed identical. See #1315 for further details.

### CODE REVIEW

**Comment by DL6ER** [comment]
Force-pushed to reword commit messages, no code changes involved.

File: src/dnsmasq_interface.c [hunk 1]
```diff
@@ -2570,95 +2570,86 @@ void print_flags(const unsigned int flags)
...
                              const char *file, const int line)
 {
+	enum reply_type new_reply = REPLY_UNKNOWN;
 	// If reply is set, we use it directly instead of interpreting the flags
 	if(reply != 0)
 	{
-		query->reply = reply;
+		new_reply = reply;
 	}
 	// else: Iterate through possible values by analyzing both the flags and the addr bits
```
[the remark this answers is not in the document]
Comment (DL6ER) [reply src/dnsmasq_interface.c:2584]: Good catch, I see `reply_NOTIMP` was never implemented. We can do this in a followup PR.

File: src/dnsmasq_interface.c [hunk 2]
```diff
@@ -2570,95 +2570,86 @@ void print_flags(const unsigned int flags)
...
 			// REFUSED query
-			query->reply = REPLY_REFUSED;
+			new_reply = REPLY_REFUSED;
 		}
 		else if(addr != NULL && addr->log.rcode == SERVFAIL)
 		{
 			// SERVFAIL query
-			query->reply = REPLY_SERVFAIL;
+			new_reply = REPLY_SERVFAIL;
 		}
```
Comment (DL6ER) [inline src/dnsmasq_interface.c:2618]: To implement `reply_NOTIMP`, we'll add here:
```suggestion
		}
		else if(addr != NULL && addr->log.rcode == NOTIMP)
		{
			// NOT IMPLEMENTED reply
			query->reply = REPLY_NOTIMP;
			new_reply = REPLY_NOTIMP;
		}
```

**Comment by DL6ER** [comment]
Confirmed working. All three numbers still match on my Pi-hole after 7 days of uninterrupted operation.
``` json
{
        "domains_being_blocked": 185134,
        "dns_queries_today": 22678,          <-------------------
        "ads_blocked_today": 2027,
        "ads_percentage_today": 8.938178,
        "unique_domains": 3613,
        "queries_forwarded": 9791,
        "queries_cached": 10633,
        "clients_ever_seen": 83,
        "unique_clients": 28,
        "dns_queries_all_types": 22678,      <-------------------
        "reply_UNKNOWN": 313,
        "reply_NODATA": 6403,
        "reply_NXDOMAIN": 214,
        "reply_CNAME": 2659,
        "reply_IP": 12272,
        "reply_DOMAIN": 288,
        "reply_RRNAME": 9,
        "reply_SERVFAIL": 0,
        "reply_REFUSED": 0,
        "reply_NOTIMP": 0,
        "reply_OTHER": 0,
        "reply_DNSSEC": 520,
        "reply_NONE": 0,
        "reply_BLOB": 0,
        "dns_queries_all_replies": 22678,    <-------------------
        "privacy_level": 0,
        "status": "enabled",
        "gravity_last_updated": {
                "file_exists": true,
                "absolute": 1648348866,
                "relative": {
                        "days": 6,
                        "hours": 9,
                        "minutes": 43
                }
        }
}
```

File: src/dnsmasq_interface.c (hunk 2 above)
Comment (yubiuser) [reply src/dnsmasq_interface.c:2618]: Added by https://github.com/pi-hole/FTL/pull/1325

**Review by yubiuser — APPROVED** [review]

File: src/dnsmasq_interface.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (yubiuser) [reply src/dnsmasq_interface.c:2584]: https://github.com/pi-hole/FTL/pull/1325

**Pull request merged**